src/evergreen/evg_config_utils.rs

```diff
@@ -203,14 +203,18 @@ impl EvgConfigUtils for EvgConfigUtilsImpl {
     ///
     /// `true` if the given task is generated.
     fn is_task_generated(&self, task: &EvgTask) -> bool {
-        task.commands.iter().any(|c| {
-            if let Function(func) = c {
-                if func.func == GENERATE_RESMOKE_TASKS {
-                    return true;
+        if let Some(commands) = &task.commands {
```

Comment:
Apparently evergreen support tasks with no command list. We need handle if that happens, so a lot of these changes are for that situation.